program/src/processor.rs

```diff
@@ -175,6 +198,26 @@ mod tests {
                 }
                 solana_program::entrypoint::SUCCESS
             }
+
+            fn sol_get_rent_sysvar(&self, var_addr: *mut u8) -> u64 {
+                unsafe {
+                    *(var_addr as *mut _ as *mut Rent) = Rent::default();
+                }
+                solana_program::entrypoint::SUCCESS
+            }
+
+            fn sol_invoke_signed(
+                &self,
+                _instruction: &Instruction,
+                account_infos: &[AccountInfo],
+                _signers_seeds: &[&[&[u8]]],
+            ) -> ProgramResult {
+                // Simulate allocate
+                unsafe {
+                    *account_infos[0].data.borrow_mut() = &mut *addr_of_mut!(DATA);
+                }
+                Ok(())
+            }
```

Comment:
This is a bit brittle if we ever add more tests, but I'm not too sure of a better way to do this, which is one of the long-standing issues with the syscall stubs.

Can you at least add a comment to `verify_with_clock` saying this function is memory-unsafe and must not be called concurrently by two different tests?

Although actually, since `violation_account_exists` also checks that the version is set, it might be ok for this syscall to no-op, and we can pass a slab of 0s as the violation account data. Did you happen to try that?

Reply:
Oh that's a really good point, this is no longer needed since we explicitly check the version #